spec/controllers/contacts_controller_spec.rb

```diff
@@ -0,0 +1,159 @@
+    it "assigns the requested contact as @contact" do
+      contact = Contact.create! valid_attributes
+      get :edit, {:id => contact.to_param}, valid_session
+      expect(assigns(:contact)).to eq(contact)
+    end
+  end
+
+  describe "POST create" do
+    describe "with valid params" do
+      it "creates a new Contact" do
+        expect {
+          post :create, {:contact => valid_attributes}, valid_session
+        }.to change(Contact, :count).by(1)
+      end
+
+      it "assigns a newly created contact as @contact" do
+        post :create, {:contact => valid_attributes}, valid_session
+        expect(assigns(:contact)).to be_a(Contact)
+        expect(assigns(:contact)).to be_persisted
+      end
+
+      it "redirects to the created contact" do
+        post :create, {:contact => valid_attributes}, valid_session
+        expect(response).to redirect_to(Contact.last)
+      end
+    end
+
+    describe "with invalid params" do
+      it "assigns a newly created but unsaved contact as @contact" do
+        post :create, {:contact => invalid_attributes}, valid_session
+        expect(assigns(:contact)).to be_a_new(Contact)
+      end
+
+      it "re-renders the 'new' template" do
+        post :create, {:contact => invalid_attributes}, valid_session
+        expect(response).to render_template("new")
+      end
+    end
+  end
+
+  describe "PUT update" do
+    describe "with valid params" do
+      let(:new_attributes) {
+        skip("Add a hash of attributes valid for your model")
+      }
+
+      it "updates the requested contact" do
+        contact = Contact.create! valid_attributes
+        put :update, {:id => contact.to_param, :contact => new_attributes}, valid_session
+        contact.reload
+        skip("Add assertions for updated state")
+      end
+
+      it "assigns the requested contact as @contact" do
+        contact = Contact.create! valid_attributes
+        put :update, {:id => contact.to_param, :contact => valid_attributes}, valid_session
+        expect(assigns(:contact)).to eq(contact)
+      end
+
+      it "redirects to the contact" do
+        contact = Contact.create! valid_attributes
+        put :update, {:id => contact.to_param, :contact => valid_attributes}, valid_session
+        expect(response).to redirect_to(contact)
+      end
+    end
+
+    describe "with invalid params" do
+      it "assigns the contact as @contact" do
+        contact = Contact.create! valid_attributes
+        put :update, {:id => contact.to_param, :contact => invalid_attributes}, valid_session
+        expect(assigns(:contact)).to eq(contact)
+      end
+
+      it "re-renders the 'edit' template" do
+        contact = Contact.create! valid_attributes
+        put :update, {:id => contact.to_param, :contact => invalid_attributes}, valid_session
+        expect(response).to render_template("edit")
+      end
+    end
+  end
+
+  describe "DELETE destroy" do
+    it "destroys the requested contact" do
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.